### PR TITLE
Update substrate

### PR DESCRIPTION
**Pull request by ukint-vs**
- `parity-scale-codec = 3.1.2`
- `scale-info = 2.0.1`
- `primitive-types = 0.11.1`

@gear-tech/dev

### CODE REVIEW

File: common/Cargo.toml [hunk 1]
```diff
@@ -6,21 +6,21 @@ edition = "2018"
 license = "GPL-3.0"
 
 [dependencies]
-codec = { package = "parity-scale-codec", version = "2", default-features = false, features = ["derive", "max-encoded-len"] }
-scale-info = { version = "1.0.0", default-features = false, features = ["derive"] }
-primitive-types = { version = "0.10.1", default-features = false, features = ["scale-info"] }
+codec = { package = "parity-scale-codec", version = "3", default-features = false, features = ["derive", "max-encoded-len"] }
```
Comment (gshep) [inline common/Cargo.toml:9]: Maybe it is better to set full version "3.1.2"?

**Review by shamilsan — APPROVED** [review]
Yeah, finally! But I vote for the full version pointing too.

File: common/Cargo.toml (hunk 1 above)
Comment (grishasobol) [reply common/Cargo.toml]: I think the better way use full version like in substrate: 3.0.0.   But it's not necessary

**Review by grishasobol — APPROVED** [review]

File: common/Cargo.toml (hunk 1 above)
Comment (ukint-vs) [reply common/Cargo.toml]: > I think the better way use full version like in substrate: 3.0.0. But it's not necessary

These dependencies are not part of substrate, so they follow SemVer rules.

**Pull request merged**